src/fhirtypes/ElementDefinition.ts

```diff
@@ -602,6 +623,301 @@ export class ElementDefinition {
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {TypeNotFoundError} when the value does not match the type of the ElementDefinition
+   */
+  fixFshQuantity(value: FshQuantity): void {
+    if (!this.hasSingleType()) {
+      throw new NoSingleTypeError('Quantity');
+    }
+    const type = this.type[0].code;
+    if (type === 'Quantity') {
+      if (this.patternQuantity) {
+        const found = this.patternQuantity;
+        const foundFshQuantity = new FshQuantity(
+          found.value,
+          new FshCode(found.code, found.system)
+        );
+        // Check if the new quantity matches the current
+        if (!value.equals(foundFshQuantity)) {
+          throw new ValueAlreadyFixedError(
+            value.toString(),
+            'Quantity',
+            foundFshQuantity.toString()
+          );
+        }
+        // if they do match, there is nothing to do, so return
+        return;
+      }
+      if (value.unit) {
+        this.fixFshCode(value.unit);
+      } else {
+        this.patternQuantity = {};
+      }
+      this.patternQuantity.value = value.value;
+    } else {
+      throw new MismatchedTypeError('Quantity', value.toString(), type);
+    }
+  }
+
+  /**
+   * Fixes a Ratio to this element.
+   * @see {@link fixValue}
+   * @param {FshRatio} value - the Ratio value to fix
+   * @throws {NoSingleTypeError} when the ElementDefinition does not have a single type
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {TypeNotFoundError} when the value does not match the type of the ElementDefinition
+   */
+  fixFshRatio(value: FshRatio): void {
+    if (!this.hasSingleType()) {
+      throw new NoSingleTypeError('Ratio');
+    }
+    const type = this.type[0].code;
+    if (type === 'Ratio') {
+      if (this.patternRatio) {
+        const found = this.patternRatio;
+        const foundNumerator = new FshQuantity(
+          found.numerator?.value,
+          new FshCode(found.numerator?.code, found.numerator?.system)
+        );
+        const foundDenominator = new FshQuantity(
+          found.denominator?.value,
+          new FshCode(found.denominator?.code, found.denominator?.system)
+        );
+        const foundRatio = new FshRatio(foundNumerator, foundDenominator);
+        // Check if the new ratio matches the old
+        if (!value.equals(foundRatio)) {
+          throw new ValueAlreadyFixedError(value.toString(), 'Ratio', foundRatio.toString());
+        }
+        // If they do match, there is nothing more to do, so return
+        return;
+      }
+      // There is no existing patternRatio, so create it
+      this.patternRatio = {};
+      this.patternRatio.numerator = {};
+      this.patternRatio.denominator = {};
+      this.patternRatio.numerator.value = value.numerator.value;
+      this.patternRatio.denominator.value = value.denominator.value;
```

Comment:
Nitpicky and probably a matter of preference, but couldn't the above 5 statements have been:
```ts
this.patternRatio = {
  numerator: { value: value.numerator.value }
  denominator: { value: value.denominator.value }
}
```